testng-core/src/main/java/org/testng/TestNG.java

```diff
@@ -317,32 +317,13 @@ private OverrideProcessor getProcessor() {
     return new OverrideProcessor(m_includedGroups, m_excludedGroups);
   }
 
-  private void parseSuite(String suitePath) {
+  private Collection<XmlSuite> parseSuite(String suitePath) {
     if (LOGGER.isDebugEnabled()) {
       LOGGER.debug("suiteXmlPath: \"" + suitePath + "\"");
     }
+    Collection<XmlSuite> allSuites = null;
```

Comment:
`allSuites` is not supposed to be null in `processCommandLineArgs `

Reply:
return empty Set